Update docs regarding new cluster wide features (#311)
restatedev/docs-restate: docs/server/clusters.mdx

This provisions the cluster with default settings specified in the server configuration. Replication settings can be updated after initial provisioning using the `restatectl config set` command, e.g. if you add more nodes to the cluster in the future.

### Cluster-wide features

At provision time, Restate enables a set of cluster-wide features by default.
The resulting feature set is persisted in the cluster's `NodesConfiguration` and **cannot be changed after provisioning**, so review it before you provision a production cluster.

<Info title="Enabling features for existing clusters">
In the future, it will be possible to enable **a subset of** features for an existing cluster.
</Info>

To opt out of a feature, pass `--disable-feature` with the feature name.
You can disable several features by repeating the flag or by passing a comma separated list:

```shell
restatectl provision --disable-feature controlled-idempotent-sharding
restatectl provision --disable-feature controlled-idempotent-sharding,another-feature
```

<Info>
By default, using an older version of `restatectl` to provision the cluster, will enable all available features.
</Info>

<Info>

>     The features that are enabled by default depend on your server version. The current default feature is `controlled-idempotent-sharding` (recommended), which limits the number of partitions keys across which service invocations are sharded

Reviewer: ```suggestion
    The features that are enabled by default depend on your server version. The current default feature is `controlled-idempotent-sharding` (recommended), which limits the number of partitions keys across which service invocations are sharded.
```

This provisions the cluster with default settings specified in the server configuration. Replication settings can be updated after initial provisioning using the `restatectl config set` command, e.g. if you add more nodes to the cluster in the future.

> ### Cluster wide features

Reviewer 2: ```suggestion
### Cluster-wide features
```

> At provision time, Restate enables a set of cluster wide features by default.

Reviewer 2: ```suggestion
At provision time, Restate enables a set of cluster-wide features by default.
```

The resulting feature set is persisted in the cluster's `NodesConfiguration` and **cannot be changed after provisioning**, so review it before you provision a production cluster.

<Info title="Enabling features for existing clusters">

> It will be possible in the future to enable **some** features for existing cluster.

Reviewer 2: ```suggestion
In the future, it will be possible to enable **a subset of** features for an existing cluster.
```

</Info>

To opt out of a feature, pass `--disable-feature` with the feature name.
You can disable several features by repeating the flag or by passing a comma separated list:

```shell
restatectl provision --disable-feature controlled-idempotent-sharding
restatectl provision --disable-feature controlled-idempotent-sharding,another-feature
```

<Info>

>  Using an older version of `restatectl` to provision the cluster will get all available features to be enabled by default.

Reviewer 2: ```suggestion
 By default, using an older version of `restatectl` to provision the cluster, will enable all available features.
```

</Info>

<Info>
    The features that are enabled by default depend on your server version. The current default feature is `controlled-idempotent-sharding`, which confines

>     idempotent invocations on unscoped services to a bounded, per-service set of partition-key buckets instead of hashing the idempotency key over the full partition-key space.

Reviewer 2: This sounds complex. Can you explain it maybe with a few simple sentences? What are the implications?

Author: it's indeed complex to explain. I was wondering if we should completely drop this section; otherwise we will have to get into too much technical details. 

restatectl already gives a brief explanation of each available cluster-wide feature, but it's not better than what is explained here.

Reviewer: Maybe something along the lines of "`controlled-idempotent-sharding`, which limits the number of partitions (or to be correct partition keys) across which service invocations are sharded.". Alternatively, not going into the details is an ok solution for me as well. 

I guess what's interesting to our users is the user impact of these features. Like why should I, as a user, want to disable this feature?

Author: I can't think of a scenario where a user would want to disable this feature.

However, users may still wonder during provisioning what this option does and whether they should leave it enabled. Some documentation explaining the feature would therefore be helpful.

I'll simplify it as @tillrohrmann suggested.

Author: done

Reviewer: @muhamadazmy did you push an update? It looks still the same. Let's try to get this PR merged because I want to merge `restate-1.7` into `main` for updating the documentation.

Author: My bad. It's pushed now.
